R/getPlot.R

```diff
@@ -0,0 +1,108 @@
+#'   - percentileStart (double) For contour plots, the percentile of the first
+#'     contour.
+#'   - percentileStep (double) For contour plots, the percentile step between
+#'     each contour.
+#'   - populationId (character) Population ID. Defaults to ungated.
+
+#'   - postSubsampleN (double) Randomly subsample the file to contain this many
+#'     events after gating.
+#'   - postSubsampleP (double) Randomly subsample the file to contain this
+#'     percent of events (0 to 1) after gating.
+#'   - renderGates (logical) Render gates into the image.
+#'   - seed (double) Seed for random number generator used for subsampling. Use
+#'     for deterministic (reproducible) subsampling. If omitted, a pseudo-random
+#'     value is used.
+#'   - smoothing (double) For density and contour plots, adjusts the amount of
+#'     smoothing. Defaults to 0 (no smoothing). Set to 1 for typical smoothing.
+#'     Higher values (up to 10) increase smoothing.
+#'   - strokeThickness (double) The thickness of histogram and contour plot
+#'     lines. Defaults to 1.
+#'   - tickLabelsQ logica Display tick labels. Defaults to false.
+#'   - ticksQ logical Display ticks. Defaults to true.
+#'   - width Integer Image width. Defaults to 228.
+#'   - xAxisLabelQ logical Display x axis label. Overrides axisLabelsQ.
+#'   - xAxisQ logical Display x axis line. Overrides axesQ.
+#'   - xTickLabelsQ logical Display x tick labels. Overrides tickLabelsQ.
+#'   - xTicksQ logical Display x ticks. Overrides ticksQ.
+#'   - yAxisLabel logical Display y axis label. Overrides axisLabelsQ.
+#'   - yAxisQ logical Display y axis line. Overrides axesQ.
+#'   - yTickLabelsQ logical Display y tick labels. Overrides tickLabelsQ.
+#'   - yTicksQ logical Display y ticks. Overrides ticksQ.
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' TODO
+#' getPlot(experimentId, fcsFileId, plotType, xChannel, yChannel)
+#' }
+getPlot <- function(experimentId,
+                    fcsFileId,
+                    plotType,
+                    xChannel,
+                    yChannel,
+                    zChannel = "",
+                    params = list(),
+                    file = "") {
+  checkDefined(experimentId)
+  experimentId <- lookupByName("experiments", experimentId)
+  checkDefined(fcsFileId)
+  fcsFileId <- lookupByName(paste("experiments", experimentId, "fcsfiles", sep = "/"), fcsFileId, "filename")
+
+  args <- list(
+    fcsFileId = fcsFileId,
+    plotType = plotType,
+    xChannel = xChannel,
+    yChannel = yChannel,
+    zChannel = zChannel
+  )
+
+  params <- c(args, params)
+
+  bytes <- baseGet(paste("experiments", experimentId, "plot", sep = "/"), params)
+
+  displayPlot <- function(b) {
+    if (requireNamespace("magick", quietly = TRUE)) {
+      magick::image_read(b)
+    } else {
+      grid::grid.raster(b)
+    }
+  }
+
+  if (nchar(file) > 0) {
+    png::writePNG(bytes, paste0(tools::file_path_sans_ext(file), ".png"))
+  } else if (.Platform$GUI == "RStudio") {
```

Comment:
We also need a way for users to get a raw image while in RStudio. Either:
* `display=TRUE` displays the plot
* `raw=FALSE` returns raw image

Reply:
:+1: